enterprise/server/workflow/service/service.go

```diff
@@ -1076,22 +1158,29 @@ func (ws *workflowService) startWorkflow(ctx context.Context, gitProvider interf
 			return err
 		}
 		invocationID := invocationUUID.String()
-		_, err = ws.executeWorkflow(ctx, apiKey, wf, wd, isTrusted, action, invocationID, nil /*=extraCIRunnerArgs*/)
-		if err != nil {
-			if err == ApprovalRequired {
-				log.CtxInfof(ctx, "Skipping workflow action %s (%s) %q (requires approval)", wf.WorkflowID, wf.RepoURL, action.Name)
-				if err := ws.createApprovalRequiredStatus(ctx, wf, wd, action.Name); err != nil {
-					log.CtxWarningf(ctx, "Failed to create workflow %s (%s) action status: %s", wf.WorkflowID, wf.RepoURL, err)
+
+		// Start executions in parallel to help reduce workflow start latency
+		// for repos with lots of workflow actions.
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			_, err = ws.executeWorkflow(ctx, apiKey, wf, wd, isTrusted, action, invocationID, nil /*=extraCIRunnerArgs*/)
+			if err != nil {
+				if err == ApprovalRequired {
+					log.CtxInfof(ctx, "Skipping workflow action %s (%s) %q (requires approval)", wf.WorkflowID, wf.RepoURL, action.Name)
+					if err := ws.createApprovalRequiredStatus(ctx, wf, wd, action.Name); err != nil {
+						log.CtxWarningf(ctx, "Failed to create workflow %s (%s) action status: %s", wf.WorkflowID, wf.RepoURL, err)
+					}
+					return
 				}
-				continue
+				// TODO: Create a UI for these errors instead of just logging on the
+				// server.
+				log.CtxWarningf(ctx, "Failed to execute workflow %s (%s) action %q: %s", wf.WorkflowID, wf.RepoURL, action.Name, err)
```

Comment:
instead of nesting the ```if err == ApprovalRequired {``` clause under ```if err != nil``` and returning early, could you separate those two clauses?

if err == ApprovalRequired {
  // blah
} else if err != nil {
  // blah blah blah
}